dbt/include/athena/macros/materializations/models/incremental/merge.sql

```diff
@@ -47,6 +47,25 @@
     {{ "," if not is_last }}
 {%- endmacro -%}
 
+
+{% macro batch_iceberg_merge(tmp_relation, target_relation, merge_part) %}
+    {% set partitions_batches = get_partition_batches(tmp_relation) %}
+    {% do log('BATCHES TO PROCESS: ' ~ partitions_batches | length) %}
+    {%- for batch in partitions_batches -%}
+        {%- do log('BATCH PROCESSING: ' ~ loop.index ~ ' OF ' ~ partitions_batches | length) -%}
+        {%- set src_batch_part -%}
+            merge into {{ target_relation }} as target
+            using (select * from {{ tmp_relation }} where {{ batch }}) as src
```

Comment:
nit: not purely related to this refactor, but I'm wondering if we can pass `src_cols_csv` and use `select {{src_cols_csv}}` instead of select *

Reply:
nice catch @nicor88 , this definitely makes sense to fix

Reply:
Yep, nice catch!
Fixed.
Please review.